### PR TITLE
[FLINK-3186] Deprecate DataSink.sortLocalOutput() methods

### PR DESCRIPTION
**Pull request by smarthi**

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java [hunk 1]
```diff
@@ -429,7 +418,6 @@ public String toString() {
 	}
 
 	public static class NestedPojo {
-		public static Object ignoreMe;
```
Comment (StephanEwen) [inline flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java]: I think this one should stay, it is there to check that the type extractor ignores static fields.

**Comment by StephanEwen** [comment]
Thanks for this contribution. A lot of good cleanup in there!

One small issue, otherwise looks good to me...

File: flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java (hunk 1 above)
Comment (smarthi) [inline flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/util/CollectionDataSets.java]: Fixed, thanks for the feedback

**Comment by StephanEwen** [comment]
Thanks for addressing the issue, merging this...

**Pull request closed without merging**